src/overmap_ui.cpp

```diff
@@ -868,9 +868,11 @@ tripoint display( const tripoint &orig, const draw_data_t &data = draw_data_t()
 
             } while( true );
 
-            if( new_note.empty() && !old_note.empty() ) {
-                // do nothing, the player should be using [D]elete
```

Comment:
Is this seriously what was in the file before this PR? Does whoever wrote this (according to Github's blame it was Codemime) just not like people?

Reply:
@codemime just moved overmap ui class to separate file

Reply:
You should blame @sparr - https://github.com/CleverRaven/Cataclysm-DDA/commit/c97b26f14163fab59735c1bfc47eb0d44e1a033b in #14447

Reply:
It seems that at that time `string_input_popup` was only a function and was not designed to distinguish confirming with empty string from canceling, so the blame ultimately goes to whoever wrote the input popup code.

Reply:
:prehistoric:

Reply:
Ah. My mistake then.

Sorry, just the way this function is structured makes it seem a little mean-spirited in terms of design. Perhaps I was overreacting.

Reply:
#14443 has a more thorough explanation. Back then, the edit dialog would return an empty string if you pressed escape, and clearing the note in that case was undesirable.